cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Gov/ProposeNewConstitution.hs

```diff
@@ -170,33 +173,48 @@ hprop_ledger_events_propose_new_constitution = integrationWorkspace "propose-new
       , "--script-file", guardRailScriptFp
       ]
 
-  minDRepDeposit <- getMinDRepDeposit epochStateView ceo
-  void $ execCli' execConfig
-    [ "conway", "governance", "action", "create-constitution"
-    , "--testnet"
-    , "--governance-action-deposit", show minDRepDeposit
-    , "--deposit-return-stake-verification-key-file", stakeVkeyFp
-    , "--anchor-url", "https://tinyurl.com/3wrwb2as"
-    , "--anchor-data-hash", proposalAnchorDataHash
-    , "--constitution-url", "https://tinyurl.com/2pahcy6z"
-    , "--constitution-hash", constitutionHash
-    , "--constitution-script-hash", constitutionScriptHash
-    , "--out-file", constitutionActionFp
-    ]
+  let relativeUrlProposal = ["ipfs", proposalAnchorDataIpfsHash]
+      relativeUrlConstitution = ["ipfs", constitutionAnchorDataIpfsHash]
 
   txbodyFp <- H.note $ work </> "tx.body"
+  minDRepDeposit <- getMinDRepDeposit epochStateView ceo
 
-  H.noteShowM_ $ waitForBlocks epochStateView 1
-  txin2 <- findLargestUtxoForPaymentKey epochStateView sbe wallet1
-
-  void $ execCli' execConfig
-    [ "conway", "transaction", "build"
-    , "--change-address", Text.unpack $ paymentKeyInfoAddr wallet1
-    , "--tx-in", Text.unpack $ renderTxIn txin2
-    , "--tx-out", Text.unpack (paymentKeyInfoAddr wallet0) <> "+" <> show @Int 5_000_000
-    , "--proposal-file", constitutionActionFp
-    , "--out-file", txbodyFp
+  -- Create temporary HTTP server with files required by the call to `cardano-cli`
+  -- In this case, the server emulates an IPFS gateway
+  serveFilesWhile
```

Comment:
What does a failure here look like?

Reply:
`serveFilesWhile` doesn't seem to interfere with the output in case of failure, you can see an example here: https://github.com/IntersectMBO/cardano-node/pull/6023/checks?check_run_id=32024584074

Is that what you meant?

Reply:
That failure is coming from line 127. Can you induce on inside the `serveFilesWhile` argument, e.g. in line 186?

Reply:
This is how it looks:
<img width="1042" alt="image" src="https://github.com/user-attachments/assets/b6b95f66-cb5a-448b-9d11-c2026a5dce9d">
It still looks like `serveFilesWhile` doesn't have an effect in the output.